docs/rustdoc/centralized_telescope/main.md

```diff
@@ -1,3 +1,55 @@
 # Telescope - Construction with Bounded DFS
-This is the main page of centralized documentation.
-It will include a generic explanation of the protocol.
+
+In scenarios where $n_p$ is small, the parameters given for prehashed construction (relative to the security parameter, $\lambda$) are not optimal, resulting in a reduced probability of constructing a valid proof in a single attempt.
+For large $n_p$, the rapid growth in potential proof tuples ensures valid ones can be found efficiently, allowing the prehashed construction to work seamlessly.
+In contrast, small $n_p$ limits the search space, making the previous parameters inadequate.
+To address this, the **construction with bounded DFS** expands on the prehashed version with *retries*, *randomized indexing*, and *bounding* the DFS.
+
+## Overview
+- When $n_p$ is large, the rapid increase in potential proof tuples enhances the chances of finding a valid proof, simplifying the construction process.
+- For small $n_p$, the element distribution across bins becomes skewed, some bins may contain multiple elements while others remain empty.
+  - It reduces the probability of finding a valid proof in a single attempt.
+- Although the average number of elements per bin remains one, this imbalance increases the completeness error. 
+  - Prehashing can still be applied, but the error becomes more pronounced.
+- To mitigate this, multiple retries indexed by $v$ are introduced, each using fresh randomization via hash functions. 
+  - It significantly lowers the completeness error and improves success rates.
+- A bounded DFS search with a predefined step limit ensures computational efficiency while maintaining a deterministic worst-case runtime.
+
+### Core components
+This generalized construction uses the same key parameters as prehashed construction but introduces:
+- Retries with index $v$: 
+  - The prover retries the proof generation process up to $r$ times.
+  - Each retry uses a different retry counter $v$, where $v \in \[1, r\]$, to diversify the search process.
+- Seeded binning:
+  - Elements in $S_p$ are prehashed into bins using $H_0(v, \cdot)$ where $v \in \[1,r\]$, grouping elements based on their hash values. We prehashed using the retry counter to get different bins for each repetition, reducing the risk of badly distributed bins.
+  - This process limits the search space for DFS, making it more efficient.
+- Bounded DFS:
+  - A depth-first search explores valid proof sequences of size $u$, using the bins for efficient lookup.
+  - The total number of steps is bounded by a limit to ensure efficiency.
+
+## Protocol
+1. **Initialization**:
+   - The prover begins by preparing for up to $r$ retries. 
+   - Each retry introduces a unique retry counter $v \in \[1, r\]$ to diversify the search space.
```

Comment:
should we use [r] here like in the prehashed doc?

Reply:
You are right; I will have a second look at both docs to see if there is an occurrence I missed.